src/VirtoCommerce.CatalogModule.Data/Search/Indexing/AggregationConverter.cs

```diff
@@ -194,6 +214,17 @@ public virtual async Task<Aggregation[]> ConvertAggregationsAsync(IList<Aggregat
             return result.ToArray();
         }
 
+        private static void RemoveRange(List<string> terms, string fieldName)
+        {
+            for (var i = terms.Count - 1; i >= 0; i--)
+            {
+                if (terms[i].StartsWith($"{fieldName}-", StringComparison.OrdinalIgnoreCase))
+                {
+                    terms.RemoveAt(i);
+                }
+            }
```

Comment:
`i` should not be decremented when we remove a term. Otherwise, we are skipping next term.